server/engine/src/main/java/org/eclipse/lsp/cobol/implicitDialects/cics/utility/CICSReadqOptionsCheckUtility.java

```diff
@@ -73,8 +73,6 @@ private void checkTd(CICSParser.Cics_readq_tdContext ctx) {
             checkHasMandatoryOptions(ctx.QUEUE(), ctx, "QUEUE");
         if (!ctx.NOSUSPEND().isEmpty())
             checkHasMandatoryOptions(ctx.TD(), ctx, "TD");
-        if (!ctx.cics_into_set().isEmpty())
```

Comment:
I think there is a deeper problem here. The grammar needs to be adjusted to only take `cics_readq_td` route when `TD` is present.
```
(other arguments ...)* TD (TD | other arguments)*
```

Reply:
or put everything in the same operand rule and sort things out in the utility.

Reply:
would this address the issue ? 07c8dde4d1dfa87bb94daf19ac08a99f8a74bf1f

Reply:
I don't think that TD has to be specified as the first operand.

Reply:
performing checks with single rule : bc79bd82de1c5b185f3c10ed9a5e66dcd72d0dda